### ISSUE TITLE
Quarkus master can't handle space in path for native mode, 1.4.1.Final can

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#8977]
Quarkus master can't handle space in path for native mode, 1.4.1.Final can handle space in path

Build generates native app but it is not working and immediatelly ends with exit code 1.

Reproducer:
```bash
rm -rf directory\ with\ spaces
mkdir directory\ with\ spaces
cd directory\ with\ spaces
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started \
    -DplatformArtifactId=quarkus-bom \
    -DclassName="org.acme.getting.started.GreetingResource" \
    -Dpath="/hello"
cd getting-started

mvn clean verify -Dnative  #FAIL
mvn clean verify -Dnative -Dquarkus.platform.version=1.4.1.Final -Dquarkus-plugin.version=1.4.1.Final # OK
```

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /Users/rsvoboda/.sdkman/candidates/java/20.0.0.r11-grl/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=UTF-8 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar getting-started-1.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http --no-server -H:-UseServiceLoaderFeature -H:+StackTrace getting-started-1.0-SNAPSHOT-runner
[getting-started-1.0-SNAPSHOT-runner:84103]    classlist:   4,022.02 ms,  1.44 GB
[getting-started-1.0-SNAPSHOT-runner:84103]        (cap):   2,818.60 ms,  1.44 GB
[getting-started-1.0-SNAPSHOT-runner:84103]        setup:   4,304.40 ms,  1.44 GB
java.lang.RuntimeException: Failed to load resource META-INF/resources/index.html
	at io.quarkus.runtime.ResourceHelper.registerResources(ResourceHelper.java:25)
	at io.quarkus.runner.AutoFeature.beforeAnalysis(AutoFeature.zig:791)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$7(NativeImageGenerator.java:674)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:63)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:674)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:530)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:445)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: java.lang.IllegalArgumentException: Failed to translate file:/Users/rsvoboda/Downloads/directory with spaces/getting-started/target/getting-started-1.0-SNAPSHOT-native-image-source-jar/getting-started-1.0-SNAPSHOT-runner.jar to local path
	at io.quarkus.runtime.util.ClassPathUtils.toLocalPath(ClassPathUtils.java:215)
	at io.quarkus.runtime.util.ClassPathUtils.readStream(ClassPathUtils.java:186)
	at io.quarkus.runtime.util.ClassPathUtils.consumeStream(ClassPathUtils.java:159)
	at io.quarkus.runtime.util.ClassPathUtils.consumeAsStreams(ClassPathUtils.java:53)
	at io.quarkus.runtime.ResourceHelper.registerResources(ResourceHelper.java:17)
	... 12 more
Caused by: java.net.URISyntaxException: Illegal character in path at index 40: file:/Users/rsvoboda/Downloads/directory with spaces/getting-started/target/getting-started-1.0-SNAPSHOT-native-image-source-jar/getting-started-1.0-SNAPSHOT-runner.jar
	at java.base/java.net.URI$Parser.fail(URI.java:2913)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3084)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3166)
	at java.base/java.net.URI$Parser.parse(URI.java:3114)
	at java.base/java.net.URI.<init>(URI.java:600)
	at java.base/java.net.URL.toURI(URL.java:1035)
	at io.quarkus.runtime.util.ClassPathUtils.toLocalPath(ClassPathUtils.java:213)
	... 16 more
[getting-started-1.0-SNAPSHOT-runner:84103]   (typeflow):   9,091.29 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]    (objects):   7,238.78 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]   (features):     372.02 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]     analysis:  17,238.98 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]     (clinit):     356.23 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]     universe:     991.22 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]      (parse):   1,545.88 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]     (inline):   2,383.63 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]    (compile):  12,152.28 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]      compile:  17,064.11 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]        image:   2,123.82 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]        write:     544.87 ms,  2.90 GB
[getting-started-1.0-SNAPSHOT-runner:84103]      [total]:  46,609.87 ms,  2.90 GB
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 49210ms
```

**Comment by geoand** [comment]
This must have been caused by: https://github.com/quarkusio/quarkus/pull/8907

**Comment by gsmet** [comment]
@geoand I let you try to fix that one for 1.4.2 as the other fix is planned for backport.

If not in time, I will drop the other commit from 1.4.2.

**Comment by geoand** [comment]
I'll take care of it no matter what :wink:
